addons/mrp/models/mrp_production.py

```diff
@@ -454,15 +454,15 @@ def _compute_state(self):
         produce and all work orders has been finished.
         """
         for production in self:
-            if not production.state:
+            if not production.state or not production.product_id:
```

Comment:
or not product_uom_id It could still traceback if you just remove the uom from the view